packages/react-experiments/src/components/UnifiedPicker/UnifiedPicker.tsx

```diff
@@ -239,212 +237,277 @@ export const UnifiedPicker = <T extends {}>(props: IUnifiedPickerProps<T>): JSX.
     setDraggedIndex(-1);
   };
 
-  const defaultDragDropEvents: IDragDropEvents = {
+  const defaultDragDropEvents = React.useRef<IDragDropEvents>({
     canDrop: _canDrop,
     canDrag: () => defaultDragDropEnabled,
     onDragEnter: _onDragEnter,
     onDragLeave: () => undefined,
     onDrop: _onDropList,
     onDragStart: _onDragStart,
     onDragEnd: _onDragEnd,
-  };
+  });
 
-  const _onKeyDown = (ev: React.KeyboardEvent<HTMLDivElement>) => {
-    // Allow the caller to handle the key down
-    if (props.onKeyDown) {
-      props.onKeyDown(ev);
-    }
+  const _onSuggestionSelected = React.useCallback(
+    (ev: any, item: IFloatingSuggestionItemProps<T>) => {
+      addItems([item.item]);
+      if (props.floatingSuggestionProps.onSuggestionSelected) {
+        props.floatingSuggestionProps.onSuggestionSelected(ev, item);
+      }
+      if (input.current) {
+        input.current.clear();
+      }
+      showPicker(false);
+    },
+    [addItems, props.floatingSuggestionProps, showPicker],
```

Comment:
I think you're supposed to pull this out from props. at least that's what lint has told me in the past
(so do the floatingSuggestionProps = props thing)

Reply:
also helps will cleaner code :)

Reply:
will do; shouldn't blindly trust the eslint rule :)

Reply:
ok, did a full pass of this file and made sure all props elements are properly destructured at the top and used in the React hook dependencies